### PR TITLE
add support for list segments endpoint

### PR DESCRIPTION
**Pull request by larrytamnjong**
Added support for List segments Endpoint #373

### CODE REVIEW

**Comment by larrytamnjong** [comment]
@marcin-krystianc

File: Consul.Test/OperatorTest.cs [hunk 1]
```diff
@@ -89,5 +89,12 @@ public async Task Operator_GetLicense()
             var queryResult = await _client.Operator.GetConsulLicense();
             Assert.NotNull(queryResult.Response);
         }
+
+        [EnterpriseOnlyFact]
+        public async Task Segment_List()
+        {
+            var segments = await _client.Operator.SegmentList();
+            Assert.IsType<string[]>(segments.Response);
```
Comment (marcin-krystianc) [inline Consul.Test/OperatorTest.cs:97]: Maybe we could check that the response is not empty (Check if the length of the array is non-zero?)

Comment (larrytamnjong) [reply Consul.Test/OperatorTest.cs]: Thanks @marcin-krystianc I have made the update

**Review by marcin-krystianc — APPROVED** [review]
👍

**Pull request merged**